lib/user.go

```diff
@@ -59,3 +125,47 @@ func (u *User) HandleConnection() error {
 	}
 	return nil
 }
+
+// HandleConnectionWithHandshake ...
+func (u *User) HandleConnectionWithHandshake() error {
+	// general args
+	C.set_user_info(u.guacUser)
+
+	// client args
+	length := int(C.get_args_length(u.guacClient.args))
+	tmpslice := (*[1 << 30]*C.char)(unsafe.Pointer(u.guacClient.args))[:length:length]
+	args := make([]string, length)
+	for i, s := range tmpslice {
+		args[i] = C.GoString(s)
+	}
+	for i := range args {
+		switch args[i] {
+		case "hostname":
+			args[i] = u.info.Host
+		case "port":
+			args[i] = u.info.Port
+		case "username":
+			args[i] = u.info.Username
+		case "password":
+			args[i] = u.info.Password
+		default:
+			args[i] = ""
+		}
+	}
+	cargs := C.makeCharArray(C.int(len(args)))
+	defer C.freeCharArray(cargs, C.int(len(args)))
+
+	if int(C.guac_client_add_user(u.guacClient, u.guacUser, C.int(len(args)), cargs)) == 0 {
+		parser := C.guac_parser_alloc()
+		C.guac_user_start(parser, u.guacUser, C.int(int(usecTimeout))) // block here
```

Comment:
This function is not defaulted export by guacamole-server, we need other means to address this issue.
I believe we have to completely reimplement `guac_user` and `guac_client`.

Note that this may reduce the performance per connection since it increases the number of cgo calls.